File: src/main/java/com/matamercer/microblog/models/repositories/LikeRepository.java
package com.matamercer.microblog.models.repositories;

import com.matamercer.microblog.models.entities.Blog;
import com.matamercer.microblog.models.entities.Like;
import com.matamercer.microblog.models.entities.Post;
import org.springframework.data.jpa.repository.JpaRepository;

public interface LikeRepository extends JpaRepository<Like, Long> {
    Like findByPost_IdAndLiker(long postId, Blog liker);
    Long countLikesByPost(Post post);
}

File: src/main/java/com/matamercer/microblog/models/repositories/LikeRepository.kt
package com.matamercer.microblog.models.repositories

import org.springframework.data.jpa.repository.JpaRepository
import com.matamercer.microblog.models.entities.Like
import com.matamercer.microblog.models.entities.Blog
import com.matamercer.microblog.models.entities.Post

interface LikeRepository : JpaRepository<Like, Long> {
    fun findByPostAndLiker(post: Post, liker: Blog): Like?
    fun countLikesByPost(post: Post): Long?
}